src/core/task/Task.ts

```diff
@@ -918,10 +925,22 @@ export class Task extends EventEmitter<TaskEvents> implements TaskLike {
 				initialStatus: this.initialStatus,
 			})
 
-			if (hasTokenUsageChanged(tokenUsage, this.tokenUsageSnapshot)) {
-				this.emit(RooCodeEventName.TaskTokenUsageUpdated, this.taskId, tokenUsage)
-				this.tokenUsageSnapshot = undefined
-				this.tokenUsageSnapshotAt = undefined
+			const now = Date.now()
+			const timeSinceLastEmit = this.lastTokenUsageEmitTime ? now - this.lastTokenUsageEmitTime : Infinity
+
+			const shouldEmitDueToThrottle = timeSinceLastEmit >= this.TOKEN_USAGE_EMIT_INTERVAL_MS
```

Comment:
Alternatively I think you could use lodash.debounce and then not need the logic for emitFinalTokenUsageUpdate